cobalt/web/environment_settings.h

```diff
@@ -46,13 +48,16 @@ class EnvironmentSettings : public script::EnvironmentSettings {
   // https://storage.spec.whatwg.org/#obtain-a-storage-key
   url::Origin ObtainStorageKey() { return url::Origin::Create(creation_url()); }
 
+  bool cors_disabled() { return cors_disabled_; }
```

Comment:
Not every Web environment has network (it's not a requirement), so there probably is a better place to make this parameter value retrievable. Perhaps in WindowOrWorkerGlobalScope, next to the CspDelegate::Options.

But actually, since this is a networking option, it seems that a great place to keep this option is something similar as what it used for HTTPSRequirement, meaning NetworkDelegate, initialized from a member of NetworkModule::Options. Even XMLHttpRequest already has access to via NetworkDelegate via NetworkModule (see XMLHttpRequestImpl::StartRequest). That way, there should be no need to add this setting to all the other various classes where you're adding it in this PR.

Reply:
This is a really better solution. The code looks much clearer now!